Mathlib/Combinatorics/SimpleGraph/Coloring.lean

```diff
@@ -308,43 +343,48 @@ theorem chromaticNumber_pos [Nonempty V] {n : ℕ} (hc : G.Colorable n) : 0 < G.
   exact Nat.not_lt_zero _ h₁
 #align simple_graph.chromatic_number_pos SimpleGraph.chromaticNumber_pos
 
-theorem colorable_of_chromaticNumber_pos (h : 0 < G.chromaticNumber) :
-    G.Colorable G.chromaticNumber := by
-  obtain ⟨h, hn⟩ := Nat.nonempty_of_pos_sInf h
+theorem colorable_of_chromaticNumber_lt_top (h : G.chromaticNumber < ⊤) :
+    G.Colorable (ENat.toNat G.chromaticNumber) := by
```

Comment:
Doesn't this work?
```suggestion
    G.Colorable G.chromaticNumber.toNat := by
```

Reply:
Unfortunately no, since `ENat.toNat` is a bundled map.

I hope Lean will one day support this.

Reply:
The issue is that `ENat.toNat : ℕ∞ →*₀ ℕ` has no `ℕ∞` argument, so dot notation can't be used.

Reply:
Then probably we should have, at least temporarily, `ENat.toNat : ℕ∞ → ℕ` and `ENat.toNatHom : ℕ∞ →*₀ ℕ`